dbms/src/Interpreters/Context.cpp

```diff
@@ -144,7 +144,7 @@ struct ContextShared
     /// Rules for selecting the compression settings, depending on the size of the part.
     mutable std::unique_ptr<CompressionCodecSelector> compression_codec_selector;
     /// Allows to remove sensitive data from queries using set of regexp-based rules
-    std::unique_ptr<SensitiveDataMasker> sensitive_data_masker;
+    std::shared_ptr<SensitiveDataMasker> sensitive_data_masker;
```

Comment:
This stopped making sense to me. Could you add a comment somewhere, that describes the relative lifetimes of the objects we have to deal with? 

In general, using weak_ptr is often somewhat harmful, because it introduces alternative execution schedules, which lead to subtle bugs. It is desirable to have one point of ownership and strictly embedded lifetimes, so that you only have one possible schedule.

Reply:
Let's go in Telegram